### ISSUE TITLE
use of "specific" in standard names

### ISSUE THREAD

**Issue by taylor13** [cf-convention/discuss#181]
Does anyone know why the word "specific" was included in the standard names land_ice_specific_mass_flux_due_to_calving and land_ice_specific_mass_flux_due_to_calving_and_ice_front_melting (but not, for example, "land_ice_runoff_flux")?

**Comment by JonathanGregory** [comment]
The word "specific" in terms relating to land ice mass balance means per unit area. It was included because "surface mass balance" is often given as an area-integral quantity e.g. in `Gt yr-1` for an ice sheet. By contrast, "specific surface mass balance" means in `kg m-2 yr-1`. Hence for clarity `specific` in included `land_ice_surface_specific_mass_balance_flux` and ...`_rate`. For consistency it was therefore also included in `land_ice_specific_mass_flux_due_to_calving[_and_ice_front_melting]` but I agree that makes them inconsistent with `land_ice_surface_melt_flux` and `land_ice_runoff_flux`, which are also in `kg m-2 s-1`. I think it would be OK to delete `specific` in `land_ice_specific_mass_flux[_due_to_calving[_and_ice_front_melting]` because these aren't customary terms anyway.

**Comment by taylor13** [comment]
I understand now why "specific" has been used.  

It might be worth revisiting the various terms related to gain and loss of land_ice.  For example, consider

`tendency_of_land_ice_mass_due_to_calving` (kg s-1) 
`land_ice_calving_rate` (m s-1)
`land_ice_lwe_calving_rate` (m s-1)
`land_ice_specific_mass_flux_due_to_calving` (kg m-2 s-1)

The first name, I think is clear.  The last refers to a flux (which throughout the standard_name list is expressed as per unit area, so "specific" is unnecessary), or it might be replaced by the easy-to-understand `tendency_of_land_ice_mass_content_due_to_calving` (where in CF , "mass_content" is consistently used to express a mass per unit area).   The other 2 names are confusing to me.   I suppose`land_ice_calving_rate`, which has units of m s-1, is a measure of the rate that the land ice thickness is changing due to calving, but calving occurs at the boundary of the ice, so would seem to change its breadth, rather than its depth.  How is this variable calculated?  I also suppose for the "lwe" varible that a liquid-water-equivalent (depth) is calculated as the volume of water in the ice that has calved spread uniformly across the grid cell (assuming water at some nominal density).  Is this correct?     This seems unnecessarily complicated, duplicative, and imprecise.  Why not in place of either/both of these just report the `tendency_of_land_ice_mass_due_to_calving` or the `tendency_of_land_ice_mass_content_due_to_calving`, which are easy to understand and make no assumptions about density of water or ice.  Thus, the number of options would be reduced from 4 to the following 2:

`tendency_of_land_ice_mass_due_to_calving` (kg s-1), which already exists, and
`tendency_of_land_ice_mass_content_due_to_calving` (kg m-2 s-1)

I would also prefer the following changes:

`land_ice_basal_specific_mass_balance_flux` (kg m-2 s-1) >> `tendency_of_land_ice_mass_content_due_to_basal_loss`

`tendency_of_land_ice_mass_due_to_basal_mass_balance` (kg s-1) >> `tendency_of_land_ice_mass
_due_to_basal_loss`

`land_ice_surface_specific_mass_balance_flux` (kg m-2 s-1) >>  `tendency_of_land_ice_mass_content_due_to_surface_gain`

`tendency_of_land_ice_mass_due_to_surface_mass_balance` (kg s-1) >> `tendency_of_land_ice_mass_due_to_surface_gain`

and I would deprecate: 
`land_ice_lwe_surface_specific_mass_balance_rate` (m s-1) and `land_ice_surface_specific_mass_balance_rate`(m s-1)

**Comment by JonathanGregory** [comment]
Dear Karl

I agree we could rename `land_ice_specific_mass_flux_due_to_calving` as `land_ice_mass_flux_due_to_calving`, without `specific`, since that would make it consistent with other standard names. This is not the name it usually has in the literature anyway, so it probably wouldn't cause any problems. Instead of `land_ice_surface|basal_specific_mass_balance_flux` we could have `land_ice_surface|basal_mass_balance_flux_per_unit_area`. We've used `per_unit_area` in other CF standard names. The terms surface and basal mass balance mean the _net_ addition of mass.

The standard names you mention have been agreed and in use for a number of years. They result from a compromise between customary terminology and consistency with CF, as is often the case, and for various reasons I don't think they'd be improved by changing the words. Also, CF doesn't prescribe what quantities people should store (principle 8 in section 1.2). CMIP does that, of course, but CF has to describe whatever data people commonly want to exchange and archive. Yes, you are correct that when a local tendency of mass change is expressed in canonical units of `m s-1` (in practice `mm yr-1` or `m yr-1`) a density has been assumed. In ice-sheet models the density is known. In reality the observation is often thickness, from altimetry, rather than mass.

Best wishes

Jonathan

**Comment by JonathanGregory** [comment]
Dear Karl @taylor13

Sorry to be confusing. I think we could rename the following three names for quantities in kg m-2 s-1:

- `land_ice_basal_specific_mass_balance_flux` &rightarrow; `land_ice_basal_mass_balance_flux`
- `land_ice_specific_mass_flux_due_to_calving` &rightarrow; `land_ice_mass_flux_due_to_calving`
- `land_ice_specific_mass_flux_due_to_calving_and_ice_front_melting` &rightarrow; `land_ice_mass_flux_due_to_calving_and_ice_front_melting`

i.e. delete `specific`. We don't need "per unit area" because that's what `flux` always means in CF standard names.

I agree with clarifying the definitions you mention. Those quantifies are treated as intensive in area in models, like snowfall and surface mass balance, for instance. Hence I would phrase the last sentence as "The measure of this rate is the mass loss per unit area per unit time divided by the density of the ice."

Best wishes

Jonathan

**Comment by taylor13** [comment]
Yes, thanks for the good summary.  We're in agreement, but I suggest we be more explicit regarding "per unit area".  I think it is per unit area of land_ice, but perhaps someone might take it to mean per unit area of the grid cell containing the ice.  

In the case of "per unit area of land_ice", how does one calculate it?  Is it (mass2-mass1)/(dt*landIceArea1) or (mass2-mass1)/(dt*landIceArea2) or (mass2-mass1)/(dt*(landIceArea1+landIceArea2)/2) or what?  where dt = the time-interval = time2-time1.  It matters quite a lot when during the time step the only remaining land_ice in a grid cell completely calves into the ocean.  If we're not careful, the ice-sheet calving rate could become infinite (undefined).  

If the area is the grid-cell area, there would be no problem because cell area does not depend on time and possibly vanish.  

Perhaps considerations of this kind would encourage reporting "tendency_of_land_ice_mass_due_to_calving", which has no area in the denominator.

**Comment by JonathanGregory** [comment]
Dear Karl @taylor13 

Since it's an intensive quantity, it "really" means the rate _at a point_, although it is typically meaned over an area. That would be a land-ice area, I agree, but implicitly such quantities only exist over land icea anyway. This is similar to sea ice thickness, for example, which is an intensive quantity that exists _at a point_ but only over sea ice. It can reported as a mean over the grid box or a mean where sea ice.

The calving flux is a bit unusual, I think, because that isn't really at a point. In reality it is across a line, and in models it is within a grid cell. But we calculate land-ice mass balance assuming all these quantities to be collocated.

Best wishes

Jonathan

**Comment by taylor13** [comment]
I agree that "flux" isn't quite the right term to use for calving loss.  I think we should have constructed names using "tendency" (as in 100's of other standard name).  So, for example, instead of ``land_ice_mass_flux_due_to_calving``, we should have defined ``tendency_of_land_ice_mass_content_due_to_calving`` (or for the opposite sign, ``minus_tendency_of_land_ice_mass_content_due_to_calving``).  Guess we could still consider that.

Regarding the area issue, I think we have never made it clear how to handle variables that only exist over a certain portion of a grid cell.  For example, ``sea_ice_temperature`` is defined only where there is sea ice, so when in ``cell_methods`` you find the phrase "area: mean", you probably should only average the temperature over the sea-ice portion of the grid cell (even if the ``cell_methods`` doesn't include "where sea_ice").  So in this case with or without "where sea_ice" you get the same area-mean.  For a variable like "sea_ice_depth", on the other hand, it might reasonably be considered as 0 where there is no sea ice, so then the area-mean would be different depending on whether or not cell method includes "where sea_ice". 

For ``land_ice_mass_flux_due_to_calving``, should we assume that the flux is zero except where land_ice exists, or should we assume that it is undefined, in which case we report the same value whether or not "where land_ice" is included in ``cell_methods``?

**Comment by JonathanGregory** [comment]
Dear Karl @taylor13 

We could rename `land_ice_specific_mass_flux_due_to_calving` (kg m-2 s-1) as `tendency_of_land_ice_amount_due_to_calving`, along the lines you suggest, where `amount` means mass per unit area, as in many standard names. Similarly for ...`calving_and_ice_front_melting`. I would support doing that. We could ask ISMIP6 for an opinion.

I agree that we have not addressed the ambiguity of means for quantities which exist only for certain area types. Perhaps that should be a separate issue.

I also think we should rename `tendency_of_change_in_land_ice_amount` (kg m-2 s-1) as `tendency_of_land_ice_amount`, which I believe is what it means, given the units i.e. it's not an acceleration.

Best wishes

Jonathan

**Comment by taylor13** [comment]
Anyone from the ISMIP community want to weigh in?

I agree with Jonathan that **_both_** calving variables should be treated similarly. 

I also agree that "change_in" should be removed from the  ``tendency_of_change_in_land_ice_amount`` because the current name is inconsistent with the specified units.
